service/history/shard/context_test.go

```diff
@@ -117,6 +118,8 @@ func (s *contextTestSuite) newContext() *contextImpl {
 		timerMaxReadLevelMap:      make(map[string]time.Time),
 		remoteClusterCurrentTime:  make(map[string]time.Time),
 		eventsCache:               eventsCache,
+		transferFailoverLevels:    make(map[string]TransferFailoverLevel),
```

Comment:
I propose creating a constructor that fills all required fields and use it in the places of usages and here. It will be easier to keep track of all required fields.

Reply:
Is `newContext()` not the constructor? 
Or you mean a separate method to create just `contextImpl` ?